stubs/Modifiers/StubsModifier.swift

```diff
@@ -115,80 +115,103 @@ class StubsModifier {
     }
 
     private func applyRule<T>(_ rule: ModificationRule, object: T) throws -> T {
+        // Traverse and apply rules recursively until the subject key to modify has been discovered.
         if let currentKey = rule.path.first {
-            if var dict = object as? [String: Any] {
-                if let dictEntry = dict[currentKey] as? [String: Any] {
-                    let modifiedEntry = try applyRule(rule.ruleByRemovingTopPathComponent(), object: dictEntry)
-                    dict[currentKey] = modifiedEntry
-                } else if let arrayEntry = dict[currentKey] as? [[String: Any]] {
-                    let newArray = try arrayEntry.compactMap {
-                        try applyRule(rule.ruleByRemovingTopPathComponent(), object: $0)
-                    }
-                    dict[currentKey] = newArray
-                } else {
-//                    printInfo("Incorrect structure of json entry\n \(dict)\n Missing key \(currentKey)")
-                    // return not modified entry. Do not interupt this process, as probably the next entry will have appropriate structure.
-                    // Such case is possible for offer response, where `facetValues` array has different structure for different facets.
-                    return dict as! T
+            guard var dict = object as? [String: Any] else {
+                throw ModificationError.incorrectExecutionPath
+            }
+
+            if let dictEntry = dict[currentKey] as? [String: Any] {
+                let modifiedEntry = try applyRule(rule.ruleByRemovingTopPathComponent(), object: dictEntry)
+                dict[currentKey] = modifiedEntry
+            } else if let arrayEntry = dict[currentKey] as? [[String: Any]] {
+                let newArray = try arrayEntry.compactMap {
+                    try applyRule(rule.ruleByRemovingTopPathComponent(), object: $0)
                 }
-                return dict as! T
+                dict[currentKey] = newArray
             } else {
-                // This shouldn't happen
-                throw ModificationError.incorrectExecutionPath
+                // Return the entry unmodified. Do not interrupt this process as the next entry may have an appropriate structure.
+                // This is possible for `Offer` responses, where the `facetValues` array has a different structure for different facets.
+            }
+
+            return dict as! T
+        }
+
+        // Replace the data structure
+        if var dict = object as? [String: Any] {
```

Comment:
I see you've moved the code from the separate methods to `applyRule`? Was there a reason for this?

Reply:
Or you probably were using the previous commit which didn't include that change.

Reply:
This commit
https://github.com/xaoc1024/Stubs/commit/0e43667ea18491ed475d1b43ee0a2a0c7fb8964f